File: Course-Registration-System-main/java-(iteration_1)/src/Course.java
import java.util.ArrayList;

//The purpose of a class is to contain information representative of a course.

public class Course {
    //Attributes required for the courses
    private String courseName;
    private String courseCode;
    private int courseSemester;
    private String courseType;
    private int credit;
    private ArrayList<CourseSection> courseSections;
    private ArrayList<Course> prerequisite;
    private Lecturer lecturer;

    //Constructor for creating a Course object with specified parameters.
    public Course(String courseName, String courseCode, int courseSemester, String courseType, int credit,
                  ArrayList<CourseSection> courseSections, ArrayList<Course> prerequisite,Lecturer lecturer) {
        super();
        this.courseName = courseName;
        this.courseCode = courseCode;
        this.courseSemester = courseSemester;
        this.courseType = courseType;
        this.credit = credit;
        this.courseSections = courseSections;
        this.prerequisite= new ArrayList<>();
        this.lecturer = lecturer;
    }

    //Getter and setter methods for each attribute.
    public String getCourseName() {
        return courseName;
    }

    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }

    public String getCourseCode() {
        return courseCode;
    }

    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public int getCourseSemester() {
        return courseSemester;
    }

    public void setCourseSemester(int courseSemester) {
        this.courseSemester = courseSemester;
    }

    public String getCourseType() {
        return courseType;
    }

    public void setCourseType(String courseType) {
        this.courseType = courseType;
    }

    public int getCredit() {
        return credit;
    }

    public void setCredit(int credit) {
        this.credit = credit;
    }

    public ArrayList<CourseSection> getCourseSections() {
        return courseSections;
    }

    public void setCourseSections(ArrayList<CourseSection> courseSections) {
        this.courseSections = courseSections;
    }

    public ArrayList<Course> getPrerequisite() {
        return prerequisite;
    }

    public void setPrerequisite(ArrayList<Course> prerequisite) {
        this.prerequisite = prerequisite;
    }

    public Lecturer getLecturer() {
       return lecturer;
    }

    public void setLecturer(Lecturer lecturer) {
       this.lecturer = lecturer;
    }

    //Information about the course is displayed.
    public String viewCourseInfo() {
        return (getCourseCode()+ "  " + getCourseName() + "\nCredit: " + getCredit()
                + "\nSemester: " + getCourseSemester() + "\nCourse Type: " + getCourseType());
    }
}

File: Course-Registration-System-main/python-(iteration_2)/src/Course.py
class Course:
    """ The purpose of this class is to contain information representative of a course. """
    
    
    """ Constructor for creating a Course object with specified parameters. """
    def __init__(self, courseName, courseCode, courseSemester, courseType, credit,
                 courseSections, prerequisite, lecturer):
        
        self.__courseName = courseName
        self.__courseCode = courseCode
        self.__courseSemester = courseSemester
        self.__courseType = courseType
        self.__credit = credit
        self.__courseSections = courseSections
        self.__prerequisite = [] 
        self.__lecturer = lecturer

    # Getter and setter methods for each attribute
    def getCourseName(self):
        return self.__courseName

    def setCourseName(self, courseName):
        self.__courseName = courseName

    def getCourseCode(self):
        return self.__courseCode

    def setCourseCode(self, courseCode):
        self.__courseCode = courseCode

    def getCourseSemester(self):
        return self.__courseSemester

    def setCourseSemester(self, courseSemester):
        self.__courseSemester = courseSemester

    def getCourseType(self):
        return self.__courseType

    def setCourseType(self, courseType):
        self.__courseType = courseType

    def getCredit(self):
        return self.__credit

    def setCredit(self, credit):
        self.__credit = credit

    def getCourseSections(self):
        return self.__courseSections

    def setCourseSections(self, courseSections):
        self.__courseSections = courseSections

    def getPrerequisite(self):
        return self.__prerequisite

    def setPrerequisite(self, prerequisite):
        self.__prerequisite = prerequisite

    def getLecturer(self):
        return self.__lecturer

    def setLecturer(self, lecturer):
        self.__lecturer = lecturer

    
    """ Information about the course is displayed. """
    def viewCourseInfo(self):
        
        return (
            f"{self.getCourseCode()}  {self.getCourseName()}\n"
            f"Credit: {self.getCredit()}\n"
            f"Semester: {self.getCourseSemester()}\n"
            f"Course Type: {self.getCourseType()}"
        )
